django/template/defaulttags.py

```diff
@@ -1498,3 +1498,44 @@ def do_with(parser, token):
     nodelist = parser.parse(("endwith",))
     parser.delete_first_token()
     return WithNode(None, None, nodelist, extra_context=extra_context)
+
+
+@register.simple_tag
+def query_string(query_dict, **kwargs):
+    """
+    Add, remove, and change parameters of the given `QueryDict`,
+    and return the result as a querystring.
+
+    For example::
+
+        {% query_string request.GET foo=3 %}
+
+    To remove a key::
+
+        {% query_string request.GET foo=None %}
+
+    To use with pagination::
+
+        {% query_string request.GET page=page_obj.next_page_number %}
+    """
+    query_dict = query_dict.copy()
+    keys_to_remove = []
+    for key, value in kwargs.items():
+        if value is None:
+            keys_to_remove.append(key)
+        try:
+            iter(value)
+        except TypeError:
+            query_dict[key] = value
+        else:
+            if isinstance(value, str):
+                query_dict[key] = value
+            else:
+                query_dict.setlist(key, value)
+    for key in keys_to_remove:
+        if key in query_dict:
+            del query_dict[key]
+    query_string = query_dict.urlencode()
+    if not query_string:
+        return ""
+    return f"?{query_string}"
```

Comment:
Does it work without `mark_safe()`? :thinking:

Reply:
It seems to, I think because `simple_tag` is doing some magic.